File: 카카오/2021카카오블라인드_4.cpp
#include<bits/stdc++.h>

#define SIZE 202
#define NON 0x3f3f3f3f

using namespace std;

int board[SIZE][SIZE];

int solution(int n, int s, int a, int b, vector<vector<int>> fares) {
    int answer = NON;

    for (int i = 1; i <= n; i++) {
        for (int j = 1; j <= n; j++) {
            if (i == j)
                board[i][j] = 0;
            else
                board[i][j] = NON;
        }
    }

    for (auto vi : fares) {
        board[vi[0]][vi[1]] = vi[2];
        board[vi[1]][vi[0]] = vi[2];
    }

    for (int k = 1; k <= n; k++) {
        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= n; j++) {
                board[i][j] = min(board[i][j], board[i][k] + board[k][j]);
            }
        }
    }

    for (int k = 1; k <= n; k++) {
        answer = min<long long>(answer, (long long)board[s][k] + board[k][a] + board[k][b]);
    }
    return answer;
}

File: 카카오/2021카카오블라인드_4.java
import java.util.Arrays;

class Solution {
    private static final int inf = 10000000;
    int[][] board = new int[201][201];

    public int solution(int n, int s, int a, int b, int[][] fares) {
        for (int[] ints : board) {
            Arrays.fill(ints, inf);
        }

        for (int i = 1; i <= n ; i++) {
            board[i][i] = 0;
        }

        for (int[] fare : fares) {
            board[fare[0]][fare[1]] = fare[2]; 
            board[fare[1]][fare[0]] = fare[2]; 
        }

        for (int k = 1; k <= n ; k++) {
            for (int i = 1; i <= n; i++) {
                for (int j = 1; j <= n; j++) {
                    if (board[i][j] > board[i][k] + board[k][j]) {
                        board[i][j] = board[i][k] + board[k][j];
                    }
                }
            }
        }

        int answer = inf;

        for (int i = 1; i <= n ; i++) {
            answer = Math.min(answer, Math.min(board[s][a] + board[s][b], board[s][i] + board[i][a] + board[i][b]));
        }

        return answer;
    }
}